// @grafbase/sdk/dist/src/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FederatedGraphConfig = exports.GraphConfig = void 0;
var auth_1 = require("./auth");
var cache_1 = require("./cache");
var operation_limits_1 = require("./operation-limits");
var experimental_1 = require("./experimental");
/**
 * Defines the complete Grafbase configuration.
 */
var GraphConfig = /** @class */ (function () {
    /** @deprecated use `graph` instead of `schema` */
    function GraphConfig(input) {
        this.graph = 'graph' in input ? input.graph : input.schema;
        if (input.auth) {
            this.auth = new auth_1.Authentication(input.auth);
        }
        if (input.operationLimits) {
            this.operationLimits = new operation_limits_1.OperationLimits(input.operationLimits);
        }
        if (input.cache) {
            this.cache = new cache_1.GlobalCache(input.cache);
        }
        if (input.experimental) {
            this.experimental = new experimental_1.Experimental(input.experimental);
        }
    }
    GraphConfig.prototype.toString = function () {
        var graph = this.graph.toString();
        var auth = this.auth ? this.auth.toString() : '';
        var operationLimits = this.operationLimits ? this.operationLimits.toString() : '';
        var cache = this.cache ? this.cache.toString() : '';
        var experimental = this.experimental ? this.experimental.toString() : '';
        return "".concat(experimental).concat(auth).concat(operationLimits).concat(cache).concat(graph);
    };
    return GraphConfig;
}());
exports.GraphConfig = GraphConfig;
var FederatedGraphConfig = /** @class */ (function () {
    function FederatedGraphConfig(input) {
        this.graph = input.graph;
        if (input.auth) {
            this.auth = new auth_1.AuthenticationV2(input.auth);
        }
        if (input.operationLimits) {
            this.operationLimits = new operation_limits_1.OperationLimits(input.operationLimits);
        }
    }
    FederatedGraphConfig.prototype.toString = function () {
        var graph = this.graph.toString();
        var auth = this.auth ? this.auth.toString() : '';
        var operationLimits = this.operationLimits ? this.operationLimits.toString() : '';
        return "".concat(auth).concat(graph).concat(operationLimits);
    };
    return FederatedGraphConfig;
}());
exports.FederatedGraphConfig = FederatedGraphConfig;
